CHANGELOG.md

```diff
@@ -1,5 +1,7 @@
 # changelog
 
+ * 2.5.2 _Oct.06.2023_
+   * [update resolver](https://github.com/iambumblehead/esmock/pull/243) to improve module resolution. See [resolvewithplus/releases/tag/v2.0.6](https://github.com/iambumblehead/resolvewithplus/releases/tag/v2.0.6)
```

Comment:
could you please add use case / bug / feature request link?

Reply:
The resolver is used by this tool https://github.com/iambumblehead/depgraph and I noticed in experiments that it was resolving cjs files where esm files should be resolved and vice-versa. I'll elaborate the comment a little bit.

Reply:
The newer bullet reads this way:

 * 2.5.2 _Oct.06.2023_
   * [update resolver](https://github.com/iambumblehead/esmock/pull/243) to improve module resolution. See [resolvewithplus/releases/tag/v2.0.6](https://github.com/iambumblehead/resolvewithplus/releases/tag/v2.0.6) and [resolvewithplus/releases/tag/v2.0.7.](https://github.com/iambumblehead/resolvewithplus/releases/tag/v2.0.7) Experiments with esmock's resolver indicated incorrect resolutions. The [spec says:](https://nodejs.org/api/packages.html#package-entry-points) _the "exports" field takes precedence over "main" in supported versions of Node.js._ The updated resolver correctly returns "main" before "exports". Also, the updated resolver reads package.json "type" to determine resolution of "import" or "require" where the older resolver sometimes returned "require" values for type "module" packages... for example, if this [inferno package]( https://www.npmjs.com/package/inferno) is updated with `{ "type": "module" }` the new resolver returns the correct "import" paths, where the old resolver does not.

Reply:
@koshic the older version of the resolver mostly had problems resolving "module" type packages but most packages are still "commonjs" for example this one https://www.npmjs.com/package/immutable?activeTab=code

Because most packages are still "commonjs" I think not many people are encountering those issues

Reply:
Sorry for delay - it looks nice, thx! If we can fix something in mixed CJS / ESM environment, we should do it )